apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/cache.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2016 Cisco Systems Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from keystoneclient import auth as ksc_auth
+from keystoneclient import session as ksc_session
+from keystoneclient.v3 import client as ksc_client
+from oslo_config import cfg
+from oslo_log import log as logging
+
+from neutron._i18n import _LW
+
+LOG = logging.getLogger(__name__)
+
+# REVISIT: We use keystone to get the name of the keystone
+# project owning each neutron resource, which by default, requires
+# admin. If we keep this, we should probably move it to a separate
+# config module. But we should also investigate whether admin is even
+# needed, or if neutron's credentials could somehow be used.
+AUTH_GROUP = 'apic_aim_auth'
+ksc_session.Session.register_conf_options(cfg.CONF, AUTH_GROUP)
+ksc_auth.register_conf_options(cfg.CONF, AUTH_GROUP)
+
+
+class ProjectNameCache(object):
```

Comment:
do we need this whole class when the Ml2 driver won't use most of it? Why not just copy the code for _get_keystone_client?

Reply:
Will remove.